### PR TITLE
Automatically deploy frontend containers to GCP for develop/master branches

### LINKED ISSUE #1180: Move vercel deployments over to google cloud on test and prod
### What would you like to be added?

We already have a Dockerfile for building frontend containers, as well as a working staging instance deployed in Google Cloud. We just have to spin up another instance linked to the `master` branch and point the DNS.

### Why is this needed?

Vercel isn't providing much value for us atm, and is actually blocking some functionality due to not supporting node.js v16 in their build containers.

### PR DESCRIPTION
**Pull request by alalonde**
## Overview

**What features/fixes does this PR include?**

This adds another workflow `gcp-deploy-frontend.yml` to our Github Actions workflows. On any push to the `develop` or `master` branch, it will automatically build a frontend container and push it to to the same URL, e.g. https://frontend-develop-mjhnbmqqna-uk.a.run.app.

Closes #1180

## Follow up tasks
- [ ] Test https://frontend-alec-develop-mjhnbmqqna-uk.a.run.app, which was the latest deployment of this branch and points to the test backend.
- [ ] Merge, push to master. Test both deployments
- [ ] If deployments are sufficient, point our DNS entries to the appropriate URLs 
- [ ] Disconnect auto-deployments for all branches (including PRs) in Vercel. Optionally disable / delete our account
- [ ] Finally merge the `blocked` PRs in this repo

We could probably do the same for the backend / hasura / DB instances as well relatively easily, I just wanted to start with something relatively simple to unblock our PRs reliant on node v16.

### CODE REVIEW

File: .github/workflows/gcp-deploy-frontend.yaml [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+          gcloud -q run deploy ${{needs.env.outputs.FRONTEND_SERVICE}} \
+            --image ${{needs.env.outputs.FRONTEND_TAG}} \
+            --region ${{needs.env.outputs.CLOUDRUN_REGION}} \
+            --port ${{needs.env.outputs.FRONTEND_PORT}} \
+            --cpu 1 \
+            --memory 512Mi \
+            --ingress all \
+            --max-instances 1 \
+            --allow-unauthenticated \
+            --set-env-vars WEB3_STORAGE_TOKEN=${{secrets.WEB3_STORAGE_TOKEN}} \
```
Comment (dysbulic) [inline .github/workflows/gcp-deploy-frontend.yaml:194]: Do we need to be exposing the YouTube API key to the frontend? *(We're not currently for these deployments though maybe it's set in Vercel.)*

**Review by dysbulic — COMMENTED** [review]
I just saw that you set `min-instances`. My remaining question is about passing a YouTube API key through from the GitHub secrets.

File: .github/workflows/gcp-deploy-frontend.yaml [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+      - name: "Deploy Container Image: ${{needs.env.outputs.FRONTEND_SERVICE}}"
+        run: |
+          gcloud -q run deploy ${{needs.env.outputs.FRONTEND_SERVICE}} \
+            --image ${{needs.env.outputs.FRONTEND_TAG}} \
+            --region ${{needs.env.outputs.CLOUDRUN_REGION}} \
+            --port ${{needs.env.outputs.FRONTEND_PORT}} \
+            --cpu 1 \
+            --memory 512Mi \
+            --ingress all \
+            --max-instances 1 \
```
[the remark this answers is not in the document]
Comment (alalonde) [reply .github/workflows/gcp-deploy-frontend.yaml:191]: sure

File: .github/workflows/gcp-deploy-frontend.yaml (hunk 1 above)
Comment (alalonde) [reply .github/workflows/gcp-deploy-frontend.yaml:194]: Not related to this PR, but I actually asked @arsenijesavic to just add it to our repo. It's not really a secret as it is used by the browser. The HTTP referer config for the API key is what secures it.

**Review by dysbulic — APPROVED** [review]
Makes sense to me…

**Pull request merged**